ci/webhook-receiver.py

```diff
@@ -158,6 +164,33 @@ def filter_to_prs_and_pr_comments(json):
 
     return None
 
+# run the manager, and rest of CI process
+def run_manager(request_ctx):
+    # requests are tuples of (context, run mode)
+    log_access_granted(request_ctx, "Running CI")
+    os.system("./manager.sh config {} \"{}\" \"{}\" {}".format(request_ctx['id'], request_ctx['repo'], request_ctx['body'], request_ctx['mode']))
+
+def request_handler():
+    global manager_running
+    while True:
+        if ci_request_list:
+            # we have a request
+            manager_running = True
+            run_manager(ci_request_list.pop(0))
+            # remove flag for other threads
+            manager_running = False
+        if not ci_request_list:
+            # list empty, go to sleep until signaled
+            request_event.wait()
+
+def add_request(request_ctx):
+    ci_request_list.append(request_ctx)
+    if not request_event.isSet():
```

Comment:
Why not just set `request_event.set()` and then do `equest_event.clear()` in the request_handler?